apps/roam/src/utils/conceptConversion.ts

```diff
@@ -0,0 +1,241 @@
+import { DiscourseNode } from "./getDiscourseNodes";
+import getDiscourseRelations from "./getDiscourseRelations";
+import type { DiscourseRelation } from "./getDiscourseRelations";
+import { Database } from "@repo/database/types.gen";
+import type { SupabaseContext } from "~/utils/supabaseContext";
+
+import type { LocalConceptDataInput } from "@repo/database/input_types";
+
+const getNodeExtraData = (
+  node_uid: string,
+): {
+  author_uid: string;
+  created: string;
+  last_modified: string;
+  page_uid: string;
+} => {
+  const result = window.roamAlphaAPI.q(
+    `[
+    :find
+      ?author_uid
+      ?page_uid
+      ?created
+      ?last_modified
+      :in $ ?block_uid
+      :where
+      [?block :block/uid ?block_uid]
+      [?block :create/user ?author_id]
+      [?author_id :user/uid ?author_uid]
+      [?block :create/time ?created]
+      [?block :edit/time ?last_modified]
+      [(get-else $ ?block :block/page ?block) ?page_id]
+      [?page_id :block/uid ?page_uid]
+  ]`,
+    node_uid,
+  );
+  if (result.length !== 1 || result[0].length !== 4)
+    throw new Error("Invalid result from Roam query");
+
+  const [author_uid, page_uid, created_t, last_modified_t] = result[0] as [
+    string,
+    string,
+    number,
+    number,
+  ];
+  const created = new Date(created_t).toISOString();
+  const last_modified = new Date(last_modified_t).toISOString();
+  return {
+    author_uid,
+    created,
+    last_modified,
+    page_uid,
+  };
+};
+
+export const discourseNodeSchemaToLocalConcept = (
+  context: SupabaseContext,
+  node: DiscourseNode,
+): LocalConceptDataInput => {
+  const titleParts = node.text.split("/");
+  return {
+    space_id: context.spaceId,
+    name: titleParts[titleParts.length - 1],
+    represented_by_local_id: node.type,
+    is_schema: true,
+    ...getNodeExtraData(node.type),
+  };
+};
+
+export const discourseNodeBlockToLocalConcept = (
+  context: SupabaseContext,
+  {
+    node_uid,
```

Comment:
On second thought, I disagree with using `_` even if they will be `_` in some other function.  This is introducing subjective interpretation on when to use `_` (like `node_uid` which is not the actual output, same with `created_t`, `last_modified_t`, etc).  But I will not block the PR further.  Just noting my thoughts.

Reply:
Actually I see it. Changing.